### PR TITLE
ROS#171 - HAL SCB Driver

### PR DESCRIPTION
**Pull request by SteelPh0enix**
Re-export of SCB driver from PAC, with integration tests. Based on #62

### CODE REVIEW

File: testbins/test-hal-scb/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    assert!(core::mem::align_of::<DummyData>() % 32 == 0);
+    assert!(core::mem::size_of::<DummyData>() % 32 == 0);
+    unsafe {
+        scb.clean_dcache_by_slice(&dummy.data);
+        scb.invalidate_dcache_by_slice(&mut dummy.data);
+    }
+
+    scb.clean_invalidate_dcache(cpuid);
+
+    // TODO: Why does this crash/halt the test?
```
Comment (Glamhoth) [inline testbins/test-hal-scb/src/main.rs:51]: I think we know why now?

File: tests/requirements/test/test_hal_scb.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Test scenario:
+// - Verify that icache can be disabled/enabled/cleared
+// - Verify that dcache can be disabled/enabled/cleared
+
+/// @SRS{ROS-FUN-BSP-SCB-020}
+/// @SRS{ROS-FUN-BSP-SCB-030}
```
Comment (Glamhoth) [inline tests/requirements/test/test_hal_scb.rs:6]: It's not covered for now, right?

**Review by Glamhoth — CHANGES_REQUESTED** [review]

File: testbins/test-hal-scb/src/main.rs (hunk 1 above)
Comment (SteelPh0enix) [reply testbins/test-hal-scb/src/main.rs]: i've changed this comment to describe our current issue better

File: tests/requirements/test/test_hal_scb.rs (hunk 2 above)
Comment (SteelPh0enix) [reply tests/requirements/test/test_hal_scb.rs]: yes, 030 is not covered, removed and added TODO for that

**Review by Glamhoth — APPROVED** [review]

**Pull request merged**